aixplain/enums/language.py

```diff
@@ -22,14 +22,59 @@
 """
 
 import logging
-
-from aixplain.utils import config
-from aixplain.utils.request_utils import _request_with_retry
+import os
+import json
+import time
+from datetime import datetime, timedelta
 from enum import Enum
 from urllib.parse import urljoin
+from aixplain.utils import config
+from aixplain.utils.request_utils import _request_with_retry
 
+CACHE_FILE = ".aixplain_cache/languages.json"
+CACHE_DURATION = timedelta(hours=24)
+
+
+def save_to_cache(languages):
+    cache_data = {
+        "timestamp": time.time(),
+        "languages": languages,
+    }
+    temp_file = CACHE_FILE + ".tmp"
+    try:
+        os.makedirs(os.path.dirname(CACHE_FILE), exist_ok=True)
+        with open(temp_file, "w") as f:
+            json.dump(cache_data, f)
+            f.flush()
```

Comment:
why we need to flush?

Reply:
I added flush because the code was cutting off the JSON file while writing, and I thought it might help. flush makes sure everything in the program’s write buffer is sent to the file right away as far as I understand, so the file is fully written before anything else happens. Again this was part of my debugging process to ensure the data was saved correctly. I'll check out if it works properly without it, but out of curiosity isn't is usually recommended to have something like this when writing to external files regardless if the function is not running into any issues? Just to make sure that the sequence of actions runs smoothly